### PR TITLE
The Cat Api Service & The Dog Api Service

### PR DESCRIPTION
**Pull request by ebiwonjumit**
Created services using https://thedogapi.com/ and https://thecatapi.com/

### CODE REVIEW

**Comment by ebiwonjumit** [comment]
PLEASE DONT REVIEW YET. STILL NEEDING API CONSUMER

**Comment by ebiwonjumit** [comment]
Good to review!

File: src/procedures/the-cat-api/the-cat-api.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+ const TheCatApi = new ApiConsumer('TheCatApi', catApiUrl, {cache: {ttl: 1}});
+ ApiConsumer.setRequiredApiKey(TheCatApi, TheCatApiKey)
+
+/**
+ * Get random cat image.
+ * @param {BreedsOfCats=} catBreeds The list of all possible Cat Breeds filterable.
+ * @returns {Image} the requested image
+ */
+TheCatApi.getRandomCatImage = async function(catBreeds = ''){
```
Comment (brollb) [inline src/procedures/the-cat-api/the-cat-api.js:25]: Isn't `catBreeds` a single breed? I was a little thrown off here initially wondering if it was a comma-separated list or something.

File: src/procedures/the-dog-api/the-dog-api.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+/**
+ * The Dogs Service provides access to the Dog API data from https://api.thedogapi.com/v1/.
+ *
+ * @service
+ * @category Society
+ */
+
+
+const logger = require('../utils/logger')('CatDog');
+const axios = require('axios');
```
Comment (brollb) [inline src/procedures/the-dog-api/the-dog-api.js:10]: unused import

File: src/procedures/the-dog-api/the-dog-api.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+const { TheDogApiKey } = require('../utils/api-key');
+const {dogTypes} = require('./types');
```
Comment (brollb) [inline src/procedures/the-dog-api/the-dog-api.js:12]: rename (see earlier comment)

File: src/procedures/the-dog-api/the-dog-api.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+// Dogs API Url
+const dogApiUrl = 'https://api.thedogapi.com/v1/images/search';
+
+const TheDogApi = new ApiConsumer('TheDogApi', dogApiUrl, {cache: {ttl: 1}});
+ApiConsumer.setRequiredApiKey(TheDogApi, TheDogApiKey)
+
+/**
+ * Get random dog image.
+ * @param {BreedsOfDogs=} dogBreeds The list of all possible Dog Breeds filterable.
```
Comment (brollb) [inline src/procedures/the-dog-api/the-dog-api.js:24]: Same comment about naming enums as in the cat api service

File: src/procedures/the-dog-api/types.js [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+    WireFoxTerrier: 259,
+    WirehairedPointingGriffon: 260,
+    WirehairedVizsla: 261,
+    Xoloitzcuintli: 262,
+    YorkshireTerrier: 264
+};
+
+function dogTypes() {
+    types.defineType({
+        name: 'BreedsOfDogs',
```
Comment (brollb) [inline src/procedures/the-dog-api/types.js:181]: See earlier comment about naming enum types

File: src/procedures/utils/api-key.js [hunk 6]
```diff
@@ -93,6 +93,18 @@ module.exports.AlphaVantageKey = new ApiKey(
     'ALPHA_VANTAGE_KEY'
 );
 
+module.exports.TheCatApiKey = new ApiKey(
+    'TheCatApi',
+    'https://thecatapi.com/signup',
+    'THE_CAT_API_KEY'
```
Comment (brollb) [inline src/procedures/utils/api-key.js:99]: This doesn't need to be set (unless there is a reason it needs this env name). This feature was initially added for backwards compatibility; the default should be good for most cases.

**Review by brollb — CHANGES_REQUESTED** [review]
Nice work, @ebiwonjumit! I added some comments below. A couple bigger comments:
- [ ] Can you add tests for the interface? These are used to ensure we don't break backwards compatibility with any existing NetsBlox projects. Examples can be found in the `test/procedures` directory.
- [ ] Add an RPC for querying the list of all supported breeds. This is useful to help users explore the RPCs before digging into the docs :)

**Pull request closed without merging**